### PR TITLE
Feature/add metrics export

### PR DESCRIPTION
**Pull request by litleleprikon**
# PR Summary

Added CLI a commands for metrics import/export:
* `--pull-trigger` and `--push-trigger`,
* `--pull-trigger-metrics` and `--push-trigger-metrics`.

Added api method `GET /api/trigger/{id}/dump` and import it to local database via CLI command `cli --push-trigger-dump --trigger-dump-file dump.json`.

Trigger dump json looks like:
```
{
    "created": "2021-03-04T07:32:54.32765027Z",
    "last_check": {
        //...
    },
    "metrics": [
        //...
    ],
    "trigger": {
        // ...
    }
}
```

### CODE REVIEW

File: cmd/cli/main.go [hunk 1]
```diff
@@ -48,6 +50,17 @@ var (
...
+	pullTrigger            = flag.String("pull-trigger", "", "Get trigger from redis and save it to file")
+	pullTriggerMetrics     = flag.String("pull-trigger-metrics", "", "Get trigger patterns and metrics from redis and save it to file")
+	pushTrigger            = flag.Bool("push-trigger", false, "Get trigger in JSON from file and save it to redis")
+	pushTriggerMetricsByID = flag.String("push-trigger-metrics", "", "Get trigger patterns and metrics in JSON from stdin and save it to redis")
+	pushTriggerDump        = flag.Bool("push-trigger-dump", false, "Get trigger dump in JSON from stdin and save it to redis")
+	triggerFile            = flag.String("trigger-file", "", "File that holds trigger JSON")
+	triggerMetricsFile     = flag.String("trigger-metrics-file", "", "File that holds trigger metrics JSON")
+	triggerDumpFile        = flag.String("trigger-dump-file", "", "File that holds trigger dump JSON from api method response")
+)
+
```
Comment (litleleprikon) [inline cmd/cli/main.go:63]: Why do we need to have all this options? As I mentioned: the use case is a simple: if something went wrong with trigger check. We just go to API and dump all data about trigger, metrics, and retentions in one JSON. Then we need only one method in CLI that will place all this data into Redis.

Comment (androndo) [reply cmd/cli/main.go]: fixed

File: api/handler/trigger.go [hunk 2]
```diff
@@ -143,3 +146,38 @@ func setTriggerMaintenance(writer http.ResponseWriter, request *http.Request) {
 		render.Render(writer, request, err) //nolint
 	}
 }
+
+func triggerDump(writer http.ResponseWriter, request *http.Request) {
+	triggerID, log := prepareTriggerContext(request)
+	response := map[string]interface{}{}
```
[the remark this answers is not in the document]
Comment (androndo) [reply api/handler/trigger.go]: fixed

File: api/controller/trigger.go [hunk 3]
```diff
@@ -163,3 +164,21 @@ func SetTriggerMaintenance(database moira.Database, triggerID string, triggerMai
...
+}
+
+//PullTrigger returns raw trigger metrics from database
+func PullTriggerMetrics(database moira.Database, logger moira.Logger, triggerID string) ([]support.PatternMetrics, *api.ErrorResponse) {
+	metrics, err := support.HandlePullTriggerMetrics(logger, database, triggerID)
+	if err != nil {
+		return nil, api.ErrorInternalServer(err)
+	}
+	return metrics, nil
+}
```
[the remark this answers is not in the document]
Comment (androndo) [reply api/controller/trigger.go]: fixed

File: api/handler/trigger.go [hunk 4]
```diff
@@ -143,3 +146,38 @@ func setTriggerMaintenance(writer http.ResponseWriter, request *http.Request) {
...
+	errors := []string{}
+
+	if trigger, err := controller.PullTrigger(database, log, triggerID); err != nil {
+		errors = append(errors, fmt.Sprintf("get_trigger_error: %s", err.ErrorText))
```
[the remark this answers is not in the document]
Comment (androndo) [reply api/handler/trigger.go]: fixed

**Review by litleleprikon — COMMENTED** [review]
Я не могу аппрувить этот ПР, так как я его открыл. @androndo апрувь его сам и сквошем вмерживай

**Review by androndo — APPROVED** [review]

**Pull request merged**